npm/ci-requirements.sh

```diff
@@ -80,13 +80,30 @@ echo "Installing curl"
 echo "Installing dependencies required for tests in codegens/shell-httpie"
 sudo apt-get install httpie
 
-echo "Installing dependencies required for tests in codegens/dart"
+echo "Installing dependencies required for tests in codegens/dart-http"
 pushd ./codegens/dart-http &>/dev/null;
-  wget -q https://storage.googleapis.com/dart-archive/channels/stable/release/2.10.2/linux_packages/dart_2.10.2-1_amd64.deb
-  sudo dpkg -i dart_2.10.2-1_amd64.deb
+    wget -q https://storage.googleapis.com/dart-archive/channels/stable/release/latest/linux_packages/dart_3.0.4-1_amd64.deb
+  sudo dpkg -i dart_3.0.4-1_amd64.deb
   echo '''name: test
+version: 1.0.0
+environment:
+  sdk: ^3.0.3
 dependencies:
-  http: ^0.12.2
+  http: ^1.0.0
+''' > pubspec.yaml
+  dart pub get
+popd &>/dev/null;
+
+echo "Installing dependencies required for tests in codegens/dart-dio"
+pushd ./codegens/dart-dio &>/dev/null;
+  wget -q https://storage.googleapis.com/dart-archive/channels/stable/release/latest/linux_packages/dart_3.0.4-1_amd64.deb
+  sudo dpkg -i dart_3.0.4-1_amd64.deb
```

Comment:
We are doing this two times, which is not required. Dart installation can be done only once, and then we can build two different `pubspec.yaml` files based on the variations.